src/Core/AdminConsole/OrganizationFeatures/Organizations/OrganizationDisableCommand.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Bit.Core.AdminConsole.OrganizationFeatures.Organizations.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Organizations;
+
+public class OrganizationDisableCommand : IOrganizationDisableCommand
+{
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IApplicationCacheService _applicationCacheService;
+
+    public OrganizationDisableCommand(
+        IOrganizationRepository organizationRepository,
+        IApplicationCacheService applicationCacheService)
+    {
+        _organizationRepository = organizationRepository;
+        _applicationCacheService = applicationCacheService;
+    }
+
+    public async Task DisableAsync(Guid organizationId, DateTime? expirationDate)
+    {
+        var organization = await _organizationRepository.GetByIdAsync(organizationId);
+        if (organization is { Enabled: true })
+        {
+            organization.Enabled = false;
+            organization.ExpirationDate = expirationDate;
+            organization.RevisionDate = DateTime.UtcNow;
+
+            await _organizationRepository.ReplaceAsync(organization);
+            await _applicationCacheService.UpsertOrganizationAbilityAsync(organization);
+
+            // TODO: send email to owners?
```

Comment:
Do we need to notify the owners?

Reply:
I left that there as it was in the original `OrganizationService` method. [Comment is from 2017](https://github.com/bitwarden/server/blob/ab0cab2072d7c925752ee7aa63125cbe2f10ed22/src/Core/AdminConsole/Services/Implementations/OrganizationService.cs#L711)